src/transformers/modeling_utils.py

```diff
@@ -2162,8 +2188,49 @@ def _get_resized_embeddings(
             dtype=old_embeddings.weight.dtype,
         )
 
-        # initialize all new embeddings (in particular added tokens)
-        self._init_weights(new_embeddings)
+        if new_num_tokens > old_num_tokens and not multivariate_resizing:
+            self._init_weights(new_embeddings)
+
+        elif new_num_tokens > old_num_tokens and multivariate_resizing:
+            # initialize new embeddings (in particular added tokens) if `new_num_tokens` is larger
+            # than `old_num_tokens`. The new embeddings will be sampled from a multivariate normal
+            # distribution that has old embeddings' mean and covariance. as described in this article:
+            # https://nlp.stanford.edu/~johnhew/vocab-expansion.html
+            logger.warning(
+                "The new embeddings will be sampled from a multivariate normal distribution that has old embeddings' mean and covariance. "
+                "As described in this article: https://nlp.stanford.edu/~johnhew/vocab-expansion.html."
+            )
+            added_num_tokens = new_num_tokens - old_num_tokens
+            if is_deepspeed_zero3_enabled() and not is_quantized:
```

Comment:
I have introduced three functions:
`self._init_added_embeddings_weights_with_mean()`
`self._init_added_lm_head_weights_with_mean()` and it uses `self._init_added_embeddings_weights_with_mean()`
`self._init_added_lm_head_bias_with_mean()`
This will improve code usability for our case. what do you think? I am open to any other change.

Also, I think that `mean_resizing` is more user-friendly and explains the whole point of the new resizing technique.